### PR TITLE
[DR-3240] Add ability to filter datasets and snapshots by id

### PR DESCRIPTION
**Pull request by samanehsan**
Update the `filter` query parameter in the enumerate datasets and enumerate snapshots endpoints to filter by a uuid.

### CODE REVIEW

**Review by s-rubenstein — APPROVED** [review]
Looks reasonable to me.

File: src/main/java/bio/terra/common/DaoUtils.java [hunk 1]
```diff
@@ -49,7 +49,9 @@ public static void addFilterClause(
     if (!StringUtils.isEmpty(filter)) {
       params.addValue("filter", DaoUtils.escapeFilter(filter));
       clauses.add(
-          String.format(" (%s.name ILIKE :filter OR %s.description ILIKE :filter) ", table, table));
+          String.format(
+              " (%s.id::text ILIKE :filter OR %s.name ILIKE :filter OR %s.description ILIKE :filter) ",
+              table, table, table));
```
Comment (snf2ye) [inline src/main/java/bio/terra/common/DaoUtils.java:54]: Oh, neat - I don't think I'm familiar with the "::" notation as a cast, but definitely a concise solution! Nice!

**Review by snf2ye — APPROVED** [review]

File: src/main/java/bio/terra/common/DaoUtils.java (hunk 1 above)
Comment (nmalfroy) [reply src/main/java/bio/terra/common/DaoUtils.java:54]: Ah, is that because it's a UUID type that you needed this?

**Review by nmalfroy — APPROVED** [review]
Looks good!  I like that you made it a `LIKE` I probably would have made it an equality but I like that your approach is more flexible (and we're at a scale where it doesn't hurt to have a wildcard operator). It does raise a possibility or a false positive (e.g. if I search for `bad`, that could theoretically be a part of a UUID, but I don't think it's a huge problem to return that)

File: src/main/java/bio/terra/common/DaoUtils.java (hunk 1 above)
Comment (samanehsan) [reply src/main/java/bio/terra/common/DaoUtils.java:54]: Yeah! I thought postgres would handle that automatically but it fails without doing this cast 🤔

**Pull request merged**